src/sync/impl/sync_metadata.hpp

```diff
@@ -76,6 +76,48 @@ class SyncUserMetadata {
     Row m_row;
 };
 
+class SyncFileActionMetadata {
+public:
+    struct Schema {
+        size_t idx_original_name;
+        size_t idx_new_name;
+        size_t idx_action;
+        size_t idx_url;
+        size_t idx_user_identity;
+    };
+
+    enum class Action {
+        // The Realm files at the given directory will be deleted.
+        DeleteRealm,
+        // The primary Realm file will be copied to a 'recovery' directory, and the original Realm files will be deleted.
+        HandleRealmForClientReset
+    };
+
+    // The absolute path to the file in question (if this is a Realm, this is the "primary" Realm file).
```

Comment:
I mean the `.realm` file itself. Is that just called the "Realm file", or does it have a particular name?